src/Tickets/Commerce/Gateways/Stripe/Status.php

```diff
@@ -163,4 +165,35 @@ public function convert_to_commerce_status( $stripe_status ) {
 
 		return tribe( Commerce_Status\Status_Handler::class )->get_by_slug( $statuses[ $stripe_status ] );
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Converts a valid Stripe payment intent to a commerce status object.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $payment_intent A Stripe payment intent.
+	 *
+	 * @return false|Commerce_Status\Status_Interface|null
+	 */
+	public function convert_payment_intent_to_commerce_status( array $payment_intent ) {
+		if ( ! isset( $payment_intent['status'] ) ) {
+			return false;
+		}
+		$stripe_status = $payment_intent['status'];
+
+		if ( ! $this->is_valid_status( $stripe_status ) ) {
+			return false;
+		}
+		$statuses = $this->get_valid_statuses();
+
+		$last_payment_error = $payment_intent['last_payment_error'] ?? null;
+
+		$has_valid_error = ! empty( $last_payment_error['decline_code'] ) || ! empty( $last_payment_error['type'] );
+
+		if ( ! $last_payment_error || ! $has_valid_error ) {
+			return tribe( Commerce_Status\Status_Handler::class )->get_by_slug( $statuses[ $stripe_status ] );
+		}
+
+		return tribe( Commerce_Status\Denied::class );
```

Comment:
Do any additional checks you need to do and then reuse the method `convert_to_commerce_status`. I cant see the reason we would have duplicate logic here. Adds a maintanability dept and easier to update one and not the other...